pkg/testing/runner/runner.go

```diff
@@ -0,0 +1,858 @@
+		return fmt.Errorf("failed to run ogc down: %w", err)
+	}
+	ps := <-proc.Wait()
+	if ps.ExitCode() != 0 {
+		return fmt.Errorf("failed to run ogc down: docker run exited with code: %d", ps.ExitCode())
+	}
+	return nil
+}
+
+// ogcMachines brings up all the instances.
+func (r *Runner) ogcMachines(ctx context.Context) ([]OGCMachine, error) {
+	var out bytes.Buffer
+	proc, err := r.ogcRun(ctx, []string{"ls", "--as-yaml"}, false, process.WithCmdOptions(attachOut(&out)))
+	if err != nil {
+		return nil, fmt.Errorf("failed to run ogc ls: %w", err)
+	}
+	ps := <-proc.Wait()
+	if ps.ExitCode() != 0 {
+		return nil, fmt.Errorf("failed to run ogc ls: docker run exited with code: %d", ps.ExitCode())
+	}
+	var machines []OGCMachine
+	err = yaml.Unmarshal(out.Bytes(), &machines)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse ogc ls output: %w", err)
+	}
+	return machines, nil
+}
+
+func (r *Runner) ogcRun(ctx context.Context, args []string, interactive bool, processOpts ...process.StartOption) (*process.Info, error) {
+	wd, err := r.getWorkDir()
+	if err != nil {
+		return nil, err
+	}
+	tokenName := filepath.Base(r.cfg.GCE.ServiceTokenPath)
+	clientEmail, err := r.cfg.GCE.ClientEmail()
+	if err != nil {
+		return nil, err
+	}
+	projectID, err := r.cfg.GCE.ProjectID()
+	if err != nil {
+		return nil, err
+	}
+	runArgs := []string{"run"}
+	if interactive {
+		runArgs = append(runArgs, "-i")
+	}
+	runArgs = append(runArgs,
+		"--rm",
+		"-e",
+		fmt.Sprintf("GOOGLE_APPLICATION_SERVICE_ACCOUNT=%s", clientEmail),
+		"-e",
+		fmt.Sprintf("GOOGLE_APPLICATION_CREDENTIALS=/root/%s", tokenName),
+		"-e",
+		fmt.Sprintf("GOOGLE_PROJECT=%s", projectID),
+		"-e",
+		fmt.Sprintf("GOOGLE_DATACENTER=%s", r.cfg.GCE.Datacenter),
+		"-v",
+		fmt.Sprintf("%s:/root/%s", r.cfg.GCE.ServiceTokenPath, tokenName),
+		"-v",
+		fmt.Sprintf("%s:%s", wd, wd),
+		"-w",
+		wd,
+		"docker.io/gorambo/ogc:blake",
+		"--",
+		"ogc",
+		"-v",
+	)
+	runArgs = append(runArgs, args...)
+	opts := []process.StartOption{process.WithContext(ctx), process.WithArgs(runArgs)}
+	opts = append(opts, processOpts...)
+	return process.Start("docker", opts...)
+}
+
+func (r *Runner) getWorkDir() (string, error) {
+	wd, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("failed to get work directory: %w", err)
+	}
+	wd, err = filepath.Abs(wd)
+	if err != nil {
+		return "", fmt.Errorf("failed to get absolute path to work directory: %w", err)
+	}
+	return wd, nil
+}
+
+func (r *Runner) mergeResults(results map[string]OSRunnerResult) (Result, error) {
+	// merge the results together
+	var rawOutput bytes.Buffer
+	var jsonOutput bytes.Buffer
+	var suites JUnitTestSuites
+	for id, res := range results {
+		batch, _ := findLayoutBatchByID(id, r.batches)
+		//if !ok {
+		//	return Result{}, fmt.Errorf("batch ID not found %s", id)
+		//}
+		batchName := fmt.Sprintf("%s/%s/%s/%s", batch.LayoutOS.OS.Type, batch.LayoutOS.OS.Arch, batch.LayoutOS.OS.Distro, batch.LayoutOS.OS.Version)
```

Comment:
maybe we can extract this loop and the one below into a function accepting a package and a prefix (for the sudo tests) ?